sogrep.in

```diff
@@ -58,23 +84,28 @@ search() {
         srepos=("${repo}")
     fi
 
+    setup_workdir
+
     for arch in "${arches[@]}"; do
         for repo in "${srepos[@]}"; do
             local prefix=
             (( VERBOSE && ${#srepos[@]} > 1 )) && prefix=${repo}/
-            db=${SOCACHE_DIR}/${arch}/${repo}/
-            if [[ -d ${db} ]]; then
+            local db=${SOCACHE_DIR}/${arch}/${repo}.links.tar.gz
+            if [[ -f ${db} ]]; then
+                local extracted=${WORKDIR}/${arch}/${repo}
+                mkdir -p "${extracted}"
+                bsdtar -C "${extracted}" -xf "${db}"
```

Comment:
why do we untar into a tempdir for each search instead of extracting once after fetching the database?

Reply:
Extracting upon download is what the current implementation does. It's too slow on my laptop HDD because it creates ~7 thousand files under `$HOME/.cache/sogrep/` -- it's even slower when it's extracting over existing files from older extractions.

The proposed change makes it temporarily extract to /tmp which is usually mounted on tmpfs and thus very fast.